<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>620</width>
    <height>799</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DBC Editor for CAN Bus Database Files</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>621</width>
      <height>781</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>DBC Editor</string>
     </attribute>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>220</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Bit Start</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_name">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>240</y>
        <width>231</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QTextEdit" name="CAN_ID">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>220</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>280</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Min</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sg_length">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>240</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_offset">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>300</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>60</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>120</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_max">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>300</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>220</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Type</string>
      </property>
     </widget>
     <widget class="QTextBrowser" name="DBC_temp">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>440</y>
        <width>571</width>
        <height>261</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Add_DBC">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>710</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Add DBC</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>280</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Unit</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="msg_cm">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>561</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_factor">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>300</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="Add_Signal">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>400</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Signal</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="msg_name">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>80</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_type">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>240</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Signed</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Unsigned</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>60</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>CAN ID</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>280</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Factor</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>280</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Max</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>220</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Endian</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>280</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Offset</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>420</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>DBC File</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Signal (EEC1)</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Del_Line">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>710</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Delete Line</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>220</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>CAN Message</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sg_start">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>240</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QSpinBox" name="msg_length">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>80</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>60</y>
        <width>48</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Name</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Add_msg">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>180</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Add Message</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Save_DBC">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>710</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Save DBC</string>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_endian">
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>240</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Little</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Big</string>
       </property>
      </item>
     </widget>
     <widget class="QTextEdit" name="sg_cm">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>360</y>
        <width>571</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_unit">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>300</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>340</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Comment</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="sg_min">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>300</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <zorder>label_4</zorder>
     <zorder>sg_name</zorder>
     <zorder>CAN_ID</zorder>
     <zorder>label_3</zorder>
     <zorder>label_14</zorder>
     <zorder>sg_length</zorder>
     <zorder>sg_offset</zorder>
     <zorder>label_10</zorder>
     <zorder>label_11</zorder>
     <zorder>sg_max</zorder>
     <zorder>label_17</zorder>
     <zorder>Add_DBC</zorder>
     <zorder>label_8</zorder>
     <zorder>msg_cm</zorder>
     <zorder>sg_factor</zorder>
     <zorder>Add_Signal</zorder>
     <zorder>msg_name</zorder>
     <zorder>sg_type</zorder>
     <zorder>label</zorder>
     <zorder>label_6</zorder>
     <zorder>label_15</zorder>
     <zorder>label_16</zorder>
     <zorder>label_7</zorder>
     <zorder>label_18</zorder>
     <zorder>label_13</zorder>
     <zorder>Del_Line</zorder>
     <zorder>label_5</zorder>
     <zorder>label_12</zorder>
     <zorder>sg_start</zorder>
     <zorder>msg_length</zorder>
     <zorder>label_2</zorder>
     <zorder>Add_msg</zorder>
     <zorder>Save_DBC</zorder>
     <zorder>sg_endian</zorder>
     <zorder>sg_cm</zorder>
     <zorder>sg_unit</zorder>
     <zorder>label_9</zorder>
     <zorder>sg_min</zorder>
     <zorder>DBC_temp</zorder>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>DBC Reader</string>
     </attribute>
     <widget class="QTextBrowser" name="txt_DBCfilepath">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>491</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>DBC File Path</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Open_DBC">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>30</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Open</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_26">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Msg List</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_27">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>360</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Signal List</string>
      </property>
     </widget>
     <widget class="QComboBox" name="sg_select_combo">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>360</y>
        <width>69</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="sg_table">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>390</y>
        <width>571</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="msg_table">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>571</width>
        <height>241</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>Recieve</string>
     </attribute>
     <widget class="QComboBox" name="sg_select_combo_2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>10</y>
        <width>69</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_24">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Data</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="live_table">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>571</width>
        <height>301</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextBrowser" name="txt_t3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>380</y>
        <width>571</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_28">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>360</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Receive Data</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
